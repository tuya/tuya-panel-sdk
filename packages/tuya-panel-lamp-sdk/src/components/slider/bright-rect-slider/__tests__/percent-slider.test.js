import React from 'react';
import { View } from 'react-native';
import { shallow } from 'enzyme';
import PercentSlider from '../index';

describe('PercentSlider components', () => {
  it('basic render', () => {
    let value = 0;
    const formatPercent = num => num;
    const wrapper = shallow(<PercentSlider value={value} formatPercent={formatPercent} />);
    expect(wrapper).toMatchSnapshot();

    value = 100;
    wrapper.setProps({ value });
  });

  it('componentWillReceiveProps render', () => {
    let value = 0;
    let direction = 'horizontal';

    const wrapper = shallow(<PercentSlider value={value} direction={direction} />);
    expect(wrapper).toMatchSnapshot();

    value = 100;
    wrapper.setProps({ value });

    value = 60;
    wrapper.setProps({ value });

    value = 10;
    wrapper.setProps({ value });

    direction = 'vertical';
    wrapper.setProps({ direction });

    // wrapper.instance().handleThumbChange(10, 10);

    wrapper.setProps({
      renderThumb: () => {
        return <View />;
      },
    });

    wrapper.unmount();
  });
});
