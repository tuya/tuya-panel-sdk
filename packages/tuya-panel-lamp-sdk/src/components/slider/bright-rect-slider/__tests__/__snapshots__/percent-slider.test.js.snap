// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PercentSlider components basic render 1`] = `
<NumberSlider
  accessibilityLabel="numberSlider"
  clickEnabled={true}
  direction="horizontal"
  disabled={false}
  formatPercent={[Function]}
  invalidSwipeDistance={7}
  max={1000}
  min={0}
  onChange={[Function]}
  onGrant={[Function]}
  onMove={[Function]}
  onPress={[Function]}
  onRelease={[Function]}
  onThumbChange={[Function]}
  reverse={false}
  showAnimation={true}
  showTint={true}
  step={1}
  style={
    Array [
      Object {
        "height": 104,
        "width": "100%",
      },
      undefined,
    ]
  }
  thumbLimitType="outer"
  thumbStyle={
    Array [
      Object {
        "opacity": 0,
      },
      undefined,
    ]
  }
  thumbTouchSize={
    Object {
      "height": 40,
      "width": 40,
    }
  }
  tintColor="#fff"
  tintStyle={
    Array [
      Object {
        "borderRadius": 0,
      },
      undefined,
    ]
  }
  trackColor="rgba(0,0,0,.4)"
  trackSlideEnabled={true}
  trackStyle={
    Array [
      Object {
        "borderRadius": 0,
        "height": 104,
      },
      undefined,
    ]
  }
  value={0}
>
  <Percent
    colorOver="#fff"
    iconColor="#000"
    iconSize={24}
    layout="left"
    length={0}
    outColor="#fff"
    percent={0}
    style={
      Array [
        Object {
          "color": "#000",
          "fontSize": 14,
        },
        undefined,
      ]
    }
  />
</NumberSlider>
`;

exports[`PercentSlider components componentWillReceiveProps render 1`] = `
<NumberSlider
  accessibilityLabel="numberSlider"
  clickEnabled={true}
  direction="horizontal"
  disabled={false}
  invalidSwipeDistance={7}
  max={1000}
  min={0}
  onChange={[Function]}
  onGrant={[Function]}
  onMove={[Function]}
  onPress={[Function]}
  onRelease={[Function]}
  onThumbChange={[Function]}
  reverse={false}
  showAnimation={true}
  showTint={true}
  step={1}
  style={
    Array [
      Object {
        "height": 104,
        "width": "100%",
      },
      undefined,
    ]
  }
  thumbLimitType="outer"
  thumbStyle={
    Array [
      Object {
        "opacity": 0,
      },
      undefined,
    ]
  }
  thumbTouchSize={
    Object {
      "height": 40,
      "width": 40,
    }
  }
  tintColor="#fff"
  tintStyle={
    Array [
      Object {
        "borderRadius": 0,
      },
      undefined,
    ]
  }
  trackColor="rgba(0,0,0,.4)"
  trackSlideEnabled={true}
  trackStyle={
    Array [
      Object {
        "borderRadius": 0,
        "height": 104,
      },
      undefined,
    ]
  }
  value={0}
>
  <Percent
    colorOver="#fff"
    iconColor="#000"
    iconSize={24}
    layout="left"
    length={0}
    outColor="#fff"
    percent={0}
    style={
      Array [
        Object {
          "color": "#000",
          "fontSize": 14,
        },
        undefined,
      ]
    }
  />
</NumberSlider>
`;
